### ISSUE TITLE
Cloudflare - 405 while deleting _acme_challenge TXT record

### ISSUE THREAD

**Issue by aaccioly** [adferrand/dnsrobocert#1017]
Hello, as stated in the title, while certificates are created as expected, but the code is failing to cleanup the challenge TXT records after successfully passing  the challenge.

Relevant lines in the log bellow (ZoneID masked):

```
dnsrobocert  | Hook '--manual-cleanup-hook' for satdomain.net reported error code 1
dnsrobocert  | Hook '--manual-cleanup-hook' for satdomain.net ran with error output:
dnsrobocert  |  Error while executing the `cleanup` hook:
dnsrobocert  |  405 Client Error: Method Not Allowed for url: https://api.cloudflare.com/client/v4/zones/eb111c0fc0000f000c0000ec0d000000/dns_records/TXT
dnsrobocert  |  Traceback (most recent call last):
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/dnsrobocert/core/hooks.py", line 40, in main
dnsrobocert  |      globals()[parsed_args.type](dnsrobocert_config, parsed_args.lineage)
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/dnsrobocert/core/hooks.py", line 123, in cleanup
dnsrobocert  |      txt_challenge(certificate, profile, token, domain, action="delete")
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/dnsrobocert/core/challenge.py", line 57, in txt_challenge
dnsrobocert  |      operations.delete_record("TXT", challenge_name, token)
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/lexicon/client.py", line 73, in delete_record
dnsrobocert  |      return self.provider.delete_record(identifier, rtype, name, content)
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/lexicon/_private/providers/cloudflare.py", line 183, in delete_record
dnsrobocert  |      self._delete(f"/zones/{self.domain_id}/dns_records/{record_id}")
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/lexicon/interfaces.py", line 187, in _delete
dnsrobocert  |      return self._request("DELETE", url, query_params=query_params)
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/lexicon/_private/providers/cloudflare.py", line 210, in _request
dnsrobocert  |      response.raise_for_status()
dnsrobocert  |    File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
dnsrobocert  |      raise HTTPError(http_error_msg, response=self)
dnsrobocert  |  requests.exceptions.HTTPError: 405 Client Error: Method Not Allowed for url: https://api.cloudflare.com/client/v4/zones/eb111c0fc0000f000c0000ec0d000000/dns_records/TXT
``` 

I've tried two different things of configuration (ZoneId, emails, domains and tokens masked):
 
1. A scoped API token (permissions Zone:Zone(read) + Zone:DNS(edit) for one zone)

    ```yaml
    draft: false
    
    acme:
      email_account: myemail@gmail.com
      staging: true
    profiles:
    - name: cloudflare_profile
      provider: cloudflare
      provider_options:
        auth_token: xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        zone_id: eb111c0fc0000f000c0000ec0d000000
    certificates:
    - domains:
      - "*.mydomain.net"
      - mydomain.net
      profile: cloudflare_profile
    ```
    
    And  API Token looks like this:  
    ![API Token](https://github.com/adferrand/dnsrobocert/assets/1591739/46f04df5-3406-413e-8eed-b83264508719)

1.  A Global API key, with –auth-username and –auth-token flags.

    ```yaml
    profiles:
    - name: cloudflare_profile
      provider: cloudflare
      provider_options:
        auth_username: myemail@gmail.com
        auth_token: yYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY
    ```

Both ways result in the same error, so I don't think that auth is a problem.

More than that, by following what [AnalogJ/lexicon](https://github.com/AnalogJ/lexicon) is doing at [cloudflare.py#L172-L183](https://github.com/AnalogJ/lexicon/blob/f8e461499389be3100b264b5b65942efbf9327bb/src/lexicon/_private/providers/cloudflare.py#L172-L183) is doing with Cloudflare's API everything works as supposed:

```shell
❯ curl -s --request GET \
  --url "https://api.cloudflare.com/client/v4/zones/eb111c0fc0000f000c0000ec0d000000/dns_records?type=TXT&name=_acme-challenge.mydomain.net" \
  --header 'Content-Type: application/json' \
  --header 'Authorization: Bearer xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' \
  | jq '.result[].id'
"bab02ba78f33352a034a98f7781534e0"
"c9716674986e3f7029f6b93a73f1a456"
❯ curl -s --request DELETE \
  --url "https://api.cloudflare.com/client/v4/zones/eb111c0fc0000f000c0000ec0d000000/dns_records/bab02ba78f33352a034a98f7781534e0" \
  --header 'Content-Type: application/json' \
  --header 'Authorization: Bearer xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
{"result":{"id":"bab02ba78f33352a034a98f7781534e0"},"success":true,"errors":[],"messages":[]}%
❯ curl -s --request DELETE \
  --url "https://api.cloudflare.com/client/v4/zones/eb111c0fc0000f000c0000ec0d000000/dns_records/c9716674986e3f7029f6b93a73f1a456" \
  --header 'Content-Type: application/json' \
  --header 'Authorization: Bearer xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
{"result":{"id":"c9716674986e3f7029f6b93a73f1a456"},"success":true,"errors":[],"messages":[]}%
```

Any ideas of what the problem may be?

**Comment by aaccioly** [comment]
Actually, I have a theory about what's going on.

At l[challenge.py#L57](https://github.com/adferrand/dnsrobocert/blob/master/src/dnsrobocert/core/challenge.py#L57) looks like `operations.delete` is called with positional parameters starting with the type and then the name.
Apparently, [`delete_record`](https://github.com/AnalogJ/lexicon/blob/f8e461499389be3100b264b5b65942efbf9327bb/src/lexicon/_private/providers/cloudflare.py#L172-L186) now expects an `identifier` as first parameter, but still falls back to the old behaviour when the `identifier` is not passed.

**Comment by aaccioly** [comment]
Fixed in [v3.24.2](https://github.com/adferrand/dnsrobocert/releases/tag/v3.24.2), relevant commits https://github.com/adferrand/dnsrobocert/commit/f1e1db5c1bd2bb427032b1c7cef184221d28c317 and https://github.com/adferrand/dnsrobocert/commit/e96572ff72aef8f8fa92eeaecf435fe7db5f88ff.
